File: exercises/python_playwright/file_upload/file_upload_playwright.py
from pathlib import Path
from playwright.sync_api import Page


class FileUploadPage:
    def __init__(self, page: Page):
        """
        Initialize locators for the upload page and embedded upload frame.
        Locators are inside iframe, thus self.iframe is used for this exercise.
        """
        self.page = page
        self.iframe = page.frame_locator("iframe")
        self.input_element = self.iframe.locator("#browse")
        self.uploaded_files = self.iframe.locator(".file-item")


    def open(self) -> None:
        """Navigate to the file upload exercise page."""
        self.page.goto("http://uitestingplayground.com/upload")


    def input_files(self, files: list[str | Path]) -> None:
        """Upload one or more files through the file input control."""
        self.input_element.set_input_files(files)


    def get_list_of_uploaded_files(self) -> list[str]:
        """Return the displayed names of the uploaded files."""
        return self.uploaded_files.all_text_contents()

File: exercises/javascript_playwright/file-upload/fileUploadPlaywright.js
export class FileUploadPage {
    /**
     *
     * @param {import('@playwright/test').Page} page
     */
    constructor(page) {
        this.page = page;
        this.iframe = page.frameLocator("iframe");
        this.fileInput = this.iframe.locator("#browse");
        this.uploadedFile = this.iframe.locator(".file-item");
    }

    async openExercisePage() {
        await this.page.goto("http://uitestingplayground.com/upload");
    }

    async uploadFiles(files) {
        await this.fileInput.setInputFiles(files);
    }

    async getUploadedFilesNames() {
        return await this.uploadedFile.allTextContents();
    }
}
